lib/puppet/provider/mongodb.rb

```diff
@@ -135,7 +135,7 @@ def self.db_ismaster
       cmd_ismaster = mongorc_file + cmd_ismaster
     end
     db = 'admin'
-    res = mongo_cmd(db, get_conn_string, cmd_ismaster).to_s.chomp()
+    res = mongo_eval(cmd_ismaster).to_s.gsub(/\n/, '').chomp()
```

Comment:
I think strip is better than gsub in most newline cases. What do you think?
"    hello    ".strip   #=> "hello"   
"\tgoodbye\r\n".strip   #=> "goodbye"